torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2146,20 +2144,13 @@ void ProcessGroupNCCL::broadcastDumpSignal() {
     // signal the monitor thread on PG0 to start dumping
     shouldDump_.store(true);
   }
-  // Give time for dumping before throwing exception
-  auto start = std::chrono::steady_clock::now();
-  auto status = promiseFlightRecorderDump_.get_future().wait_for(
+  // Give time for dumping before throwing exception for all ranks.
+  // It is hard to presume or control what the pattern of watchdog might look
+  // like, so it is better to let all ranks universally sleep for a short period
+  // of time, in this case, 60 seconds, which is also the maximum time we leave
+  // for FR dump.
+  std::this_thread::sleep_for(
       std::chrono::milliseconds(waitTimeoutDumpInMilSec_));
```

Comment:
You may want to put the sleep somewhere else than inside the `broadcastDumpSignal` function, so that others calling this function would be sure what it does. Like, one-level up maybe?

Reply:
btw, do we need "all ranks" to sleep? Can it be just the rank that's throwing exception?

Reply:
sounds good to me.